api/src/db/destinations_pipelines.rs

```diff
@@ -39,7 +34,7 @@ pub enum DestinationPipelinesDbError {
 
 #[expect(clippy::too_many_arguments)]
 pub async fn create_destination_and_pipeline(
-    pool: &PgPool,
+    txn: &mut Transaction<'_, Postgres>,
```

Comment:
`Transaction<'_, Postgres>` can be replaced with a tiny bit shorter `PgTransaction<'c>`